packages/ui/src/components/DatePicker/SDatePicker.vue

```diff
@@ -0,0 +1,734 @@
+    this.value = date
+  }
+}
+
+const customInputLength = computed(() => {
+  return props.time ? 17 : 10
+})
+
+// #endregion
+
+// #region FORMATTING
+
+const formatPattern = computed(() => {
+  return props.time ? 'dd/MM/yyyy, HH:mm' : 'dd/MM/yyyy'
+})
+
+const formatDate = (date: any) => {
+  if (!date) return ''
+  // return DateTime.fromJSDate(date).toFormat(formatPattern.value)
+  return format(date, formatPattern.value)
+}
+
+const fromFormat = (dateString: string) => {
+  // let date = DateTime.fromFormat(dateString, formatPattern.value)
+  let date = parse(dateString, formatPattern.value, new Date())
+  if (!isValid(date)) return null
+  console.dir(date)
+  return date
+}
+
+// #endregion
+
+// #region POPPER
+const [showPopper, togglePopper] = useToggle(false)
+whenever(and(props.disabled, showPopper), () => togglePopper(false), { immediate: true })
+
+const updateShow = () => {
+  if (!props.disabled) togglePopper(true)
+}
+// #endregion
+
+if (props.modelValue || (props.modelValue as Date[])?.length > 0) init()
+else updateModelValue()
+</script>
+
+<template>
+  <div class="date-picker-component">
+    <SPopover
+      v-model:show="showPopper"
+      placement="bottom-start"
+      trigger="manual"
+      distance="4"
+      @click-outside="togglePopper(false)"
+    >
+      <template #trigger>
+        <div
+          class="head-title p-2 sora-tpg-ch2 relative pr-6"
+          :class="disabled ? 'cursor-default' : 'cursor-pointer'"
+          @keydown="updateShow"
+          @click="updateShow"
+        >
+          {{ headTitle || 'Date' }}
+          <div
+            class="head-title__arrow"
+            :class="arrowState"
+          >
+            <IconArrowsChevronBottom24 />
+          </div>
+        </div>
+      </template>
+      <template #popper>
+        <SPopoverWrappedTransition
+          name="s-select-dropdown-transition"
+          eager
+        >
+          <div
+            class="date-picker sora-tpg-p4"
+            data-testid="date-picker"
+            :class="[`${gridType}`, { 'narrow': showStateView }]"
+          >
+            <div
+              v-if="props.type != 'pick'"
+              class="options-panel sora-tpg-p3"
+            >
+              <OptionsWrapper v-slot="{ options }">
+                <p
+                  v-for="(item, idx) in options[props.type]"
+                  :key="idx"
+                  class="options-panel__item"
+                  :class="menuState === item.label ? 'active' : ''"
+                  @click="onMenuClick(item.value, item.label)"
+                  @keydown="onMenuClick(item.value, item.label)"
+                >
+                  {{ item.label }}
+                  <span
+                    v-show="menuState === item.label"
+                    class="options-panel__checkmark"
+                  ><IconBasicCheckMark24 /></span>
+                </p>
+              </OptionsWrapper>
+            </div>
```

Comment:
An interesting outcome of using BEM with Vue SFC is that when you **need** to use another BEM element within the component, usually it means you should move it out to a new component.

Thus, I ask you to extract `OptionPanel` as separate component. And don't forget about components naming - I think it's name might be like `SDatePickerOptionPanel`.